MYSQL is a typedef name, if replace st_mysql with MYSQL, the compiler will say
```
In file included from /data/sqlpp11/sqlpp11-connector-mysql/src/connection.cpp:33:0:
/data/sqlpp11/sqlpp11-connector-mysql/include/sqlpp11/mysql/connection.h:39:8: error: using typedef-name ‘MYSQL’ after ‘struct’
 struct MYSQL;
        ^~~~~
In file included from /data/sqlpp11/sqlpp11-connector-mysql/src/detail/../sqlpp_mysql.h:30:0,
                 from /data/sqlpp11/sqlpp11-connector-mysql/src/detail/connection_handle.h:31,
                 from /data/sqlpp11/sqlpp11-connector-mysql/src/connection.cpp:29:
/usr/include/mysql/mysql.h:314:3: note: ‘MYSQL’ has a previous declaration here
 } MYSQL;
   ^~~~~
make[2]: *** [src/CMakeFiles/sqlpp-mysql.dir/connection.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/sqlpp-mysql.dir/all] Error 2
make: *** [all] Error 2
```

--- a/include/sqlpp11/mysql/connection.h
+++ b/include/sqlpp11/mysql/connection.h
@@ -36,6 +36,12 @@
 #include <sstream>
 #include <string>
 
+#if LIBMYSQL_VERSION_ID < 80000
+typedef struct st_mysql MYSQL;
+#else
+struct MYSQL;
+#endif
+
 namespace sqlpp
 {
   namespace mysql
@@ -294,6 +300,8 @@ namespace sqlpp
 
       //! report a rollback failure (will be called by transactions in case of a rollback failure in the destructor)
       void report_rollback_failure(const std::string message) noexcept;
+
+      MYSQL* get_handle();
     };
 
     inline std::string serializer_t::escape(std::string arg)

--- a/src/connection.cpp
+++ b/src/connection.cpp
@@ -280,5 +280,9 @@ namespace sqlpp
     {
       std::cerr << "MySQL message:" << message << std::endl;
     }
+
+    MYSQL* connection::get_handle(){
+      return _handle->mysql.get();
+    }
   }
 }

